docs/modules/ROOT/pages/account-abstraction.adoc

```diff
@@ -0,0 +1,194 @@
+
+==== Alternative signature schemes
+
+Users may add different signature validation schemes such as xref:api:utils.adoc#P256[`P256`] or xref:api:utils.adoc#RSA[`RSA`] to their accounts in a similar fashion to ECDSA:
+
+```solidity
+include::api:example$account/MyAccountP256PlainSigner.sol[]
+```
+
+=== Modules
+
+Modules are a way to extend functionality of an smart account. Given the variety of smart account implementations, a common approach has been to enable a system of modules to which accounts can delegate logic. As a result, the community has proposed xref:erc7579.adoc#ERC7579[ERC-7579] as a minimal generalized approach to smart account modules.
+
+OpenZeppelin's xref:api:account.adoc#AccountERC7579[`AccountERC7579`] is an ERC-7579 compliant implementation that works without a signer, and instead, uses a validator module installed on the account. This validator module might be any of the https://erc7579.com/modules[validators developed by the community] or one of the xref:api:account.adoc#validators[validator modules we provide].
+
+To setup a modular Account, start by importing xref:api:account.adoc#AccountERC7579[`AccountERC7579`] and make sure to install a module on its initialization.
+
+```solidity
+include::api:example$account/MyModularAccount.sol[]
+```
+
+WARNING: An account that doesn't setup a module on deployment will be unusable if there's no other execution method enabled on the account.
+
+==== Using with a signer
+
+A modular account can use a signer too. It just needs to override `_validateUserOp` logic to use the signer as part of the validation phase:
+
+```solidity
+include::api:example$account/MyModularAccountECDSATypedSigner.sol[]
+```
+
+== Account Factory
+
+Accounts are deployed using an xref:api:account.adoc#AccountFactory[`AccountFactory`] contract, a special contract that creates new accounts during the validation phase of the EntryPoint. 
+
+=== Clones
+
+So far this section has shown how to setup an account as an individual contract. However, given that a single factory will deploy the same account multiple times with different arguments, it's recommended to use the xref:api:utils.adoc#Clones[`Clones`] library to produce minimal cloned versions that are cheap to deploy.
+
+The following example demonstrates this concept with an xref:api:account.adoc#AccountERC7579[`AccountERC7579`]
+
+```solidity
+include::api:example$account/MyModularAccountClonable.sol[]
+```
+
+TIP: xref:upgradeable.adoc[OpenZeppelin Contracts upgradeable] are initializable out of the box.
+
+NOTE: For upgradeable accounts, you might want to check out the xref:api:proxy.adoc#BeaconProxy[`BeaconProxy`] pattern to have a single implementation contract from which upgradeable clones can be created.
```

Comment:
I believe beacon proxies will run into validation rule violations so it is not a good recommendation.